client/main.go

```diff
@@ -59,21 +60,21 @@ type UserHandleFunc func(*bufio.Writer, string)
 func handleServerInput(reader *bufio.Reader, handlers map[scs.ActionName]ServerHandleFunc, state *State) {
 	for {
 		action := &scs.Action{}
-		err := gob.NewDecoder(reader).Decode(action)
-		switch {
-		case err == io.EOF:
-			log.Println("Reached EOF - close this connection.\n   ---")
-			return
-		case err != nil:
-			log.Println("\nError reading command.", err)
+		if err := gob.NewDecoder(reader).Decode(action); err != nil {
+			if err == io.EOF {
+				log.Println("Reached EOF - close this connection.\n   ---")
```

Comment:
Return early when you can, indenting the error flow. Also, use `log.Printf` to print the err.

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

```go
if err == io.EOF {
	log.Println("Reached EOF - close this connection.\n   ---")
	return
}
log.Printf("\nError reading command: ", err) // I assume you want the extra newline here.
return
```